code/modules/mining/boulder_processing/refinery.dm

```diff
@@ -55,36 +45,43 @@
 	name = "boulder smelter"
 	desc = "BS for short. Accept boulders and refines metallic ores into sheets."
 	icon_state = "smelter"
-	processable_materials = list(
-		/datum/material/iron,
-		/datum/material/titanium,
-		/datum/material/silver,
-		/datum/material/gold,
-		/datum/material/uranium,
-		/datum/material/mythril,
-		/datum/material/adamantine,
-		/datum/material/runite,
-	)
 	light_system = OVERLAY_LIGHT
-	light_range = 1
-	light_power = 2
+	light_range = 2
+	light_power = 3
 	light_color = "#ffaf55"
-	light_on = FALSE
 	circuit = /obj/item/circuitboard/machine/smelter
 	usage_sound = 'sound/machines/mining/smelter.ogg'
+	action = "smelting"
 
-/obj/machinery/bouldertech/refinery/smelter/RefreshParts()
+/obj/machinery/bouldertech/refinery/smelter/Initialize(mapload)
 	. = ..()
-	light_power = boulders_processing_max
+	set_light_on(TRUE)
+
+/obj/machinery/bouldertech/refinery/smelter/can_process_material(datum/material/possible_mat)
+	var/static/list/processable_materials
+	if(!length(processable_materials))
+		processable_materials = list(
+			/datum/material/iron,
+			/datum/material/titanium,
+			/datum/material/silver,
+			/datum/material/gold,
+			/datum/material/uranium,
+		)
+	return is_type_in_list(possible_mat, processable_materials)
 
-/obj/machinery/bouldertech/refinery/smelter/accept_boulder(obj/item/boulder/new_boulder)
+/obj/machinery/bouldertech/refinery/smelter/set_light_on(new_value)
+	if(panel_open || !anchored || !is_operational || machine_stat & (BROKEN | NOPOWER))
+		new_value = FALSE
+	return ..(new_value)
```

Comment:
```suggestion
	return ..()
```
Pretty sure it passes down modified values too. Tell me if I'm wrong.

Reply:
Yes true but Fikou taught me to be explicit when overriding values of procs 
https://github.com/tgstation/tgstation/pull/80179#discussion_r1419790226
Either way no harm done so I'll follow your suggestion anyway

Reply:
The problem with being explicit here is that... let's say, you've a proc with args `user`, `target` `harm` `modifiers`, and  an override of that same proc where harm is always set to FALSE. You cannot simply call `..(harm = FALSE)` because doing so will disregard the other args. So you either perform `..(user, target, harm = FALSE, modifiers)` or just make sure that `harm` is set to false before `..()`. You can guess which one is better for long-term maintainability.